zipline/pipeline/loaders/blaze/core.py

```diff
@@ -1104,6 +1104,11 @@ def collect_expr(e, lower):
             materialized_deltas,
             dates,
         )
+        # If we ever have cases where we find out about multiple asof_dates'
+        # data on the same TS, we want to make sure that last_in_date_group
+        # selects the correct last asof_date's value.
+        sparse_output.sort_values(AD_FIELD_NAME, inplace=True)
```

Comment:
We could add some conditional logic to sort the input on some specified fields. 

My thought here was that `last_in_date_group` should just expect data that's already sorted so that all it needs to do is to group and select that last item in each group, as the name of the method suggests.